### PR TITLE
Enhancement on Initial Claims functionality

### PR DESCRIPTION
**Pull request by its-kios09**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

- Added Responsiveness of the claims page
- Added Metrics to provide Claims summary
- Added Filtering of the line items by PAID bill status
- Added a Custom Hook for fetching the providers and encounters
- Added Inline Loading and Inline notification for the end-user experience
- Added a functionality after submission to navigate back to the billing dashboard and a function to extract the name of the provider for better experience of the end-user
- Added validation of the form all fields are required
- Added itemOrBillableServiceConcept_id on the payload

## Screenshots

https://github.com/palladiumkenya/kenyaemr-esm-3.x/assets/67967749/cb4dc40d-bf40-4c1c-b1f1-0cb7276a21c8

## Other

*None.*

### CODE REVIEW

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx [hunk 1]
```diff
@@ -105,64 +112,114 @@ const ClaimsForm: React.FC<ClaimsFormProps> = ({ bill }) => {
...
+      insurer: 'SHA',
+      billNumber: billUuid,
     };
+    try {
+      await processClaims(payload);
+      const timeoutId = setTimeout(clearNotification, 5000);
+      setNotification({
+        kind: 'success',
+        title: t('processClaim', 'Process Claim'),
+        subtitle: t('sendClaim', 'Claim sent successfully'),
```
Comment (makombe) [inline packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx:172]: We can change the notification here to use snackbar for uniformity purposes

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx [hunk 2]
```diff
@@ -54,49 +58,52 @@ const ClaimsFormSchema = z.object({
...
+const ClaimsForm: React.FC<ClaimsFormProps> = ({ bill, selectedLineItems }) => {
   const { t } = useTranslation();
   const { mflCodeValue } = useSystemSetting('facility.mflcode');
   const { patientUuid, billUuid } = useParams();
   const { visits: recentVisit } = useVisit(patientUuid);
+  const visitUuid = recentVisit?.visitType.uuid;
+
+  const { data } = useProviders();
+  const [providers, setProviders] = useState([]);
+  const [notification, setNotification] = useState({ kind: '', title: '', subtitle: '', timeoutId: null });
```
Comment (makombe) [inline packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx:70]: Remove this to and use snackbar

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx [hunk 3]
```diff
@@ -105,64 +112,114 @@ const ClaimsForm: React.FC<ClaimsFormProps> = ({ bill }) => {
...
+        });
+      }, 2000);
+    } catch (err) {
+      console.error(err);
+      const timeoutId = setTimeout(clearNotification, 3000);
+      setNotification({
+        kind: 'error',
+        title: t('claimError', 'Claim Error'),
+        subtitle: t('sendClaimError', 'Request Failed, Please try later........'),
+        timeoutId,
```
Comment (makombe) [inline packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx:188]: same here use snackbar

File: packages/esm-billing-app/src/root.component.tsx [hunk 4]
```diff
@@ -13,6 +13,7 @@ const RootComponent: React.FC = () => {
         <Route path="/" element={<BillingDashboard />} />
         <Route path="/patient/:patientUuid/:billUuid" element={<Invoice />} />
         <Route path="/patient/:patientUuid/:billUuid/claims" element={<ClaimScreen />} />
+        <Route path="/patient/:patientUuid/:billUuid/claims" element={<ClaimScreen />} />
```
Comment (makombe) [inline packages/esm-billing-app/src/root.component.tsx:16]: I can see repetition here, Please remove one

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx (hunk 2 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx:70]: sorted

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx (hunk 3 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx]: sorted

File: packages/esm-billing-app/src/root.component.tsx (hunk 4 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/root.component.tsx]: sorted

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx (hunk 3 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx]: sorted

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx (hunk 1 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx]: sorted

**Comment by donaldkibet** [comment]
@its-kios09 please fix the failing build.

**Comment by its-kios09** [comment]
working on the building issues

**Comment by its-kios09** [comment]
Unclean branch

**Pull request closed without merging**